### DISCUSSION TITLE
PySyft won't work in Google Colab

### DISCUSSION THREAD

**Discussion by desertnaut** [OpenMined/courses#291] [Course: Foundations of Private Computation]
I was hoping to follow the Foundations course using Google Colab, but unfortunately this seems impossible due to the version of Jupyter notebook in Google Colab being older than 6 (in my region, at least).

I successfully installed PySyft and the course materials according to the instructions:
```
!pip install git+https://github.com/OpenMined/PySyft@dev#egg=syft
!git clone https://github.com/OpenMined/courses.git --branch foundations-of-private-computation --single-branch
```

but when trying to

```python
import syft as sy
```

I get

```python
---------------------------------------------------------------------------

Exception                                 Traceback (most recent call last)

<ipython-input-2-281ee5c8ca5f> in <module>()
----> 1 import syft as sy

2 frames

/usr/local/lib/python3.7/dist-packages/syft/core/common/environment.py in <module>()
     17     if NOTEBOOK_VERSION < version.parse("6.0.0"):
     18         raise Exception(
---> 19             "Your Jupyter Notebook is too old. Please upgrade to version 6 or higher."
     20         )
     21 except NameError:

Exception: Your Jupyter Notebook is too old. Please upgrade to version 6 or higher.
```
Indeed, `!jupyter --version` gives
```
jupyter core     : 4.5.0
jupyter-notebook : 5.2.2
qtconsole        : 4.5.2
ipython          : 5.5.0
ipykernel        : 4.10.1
jupyter client   : 5.3.1
jupyter lab      : not installed
nbconvert        : 5.5.0
ipywidgets       : 7.5.0
nbformat         : 4.4.0
traitlets        : 4.3.2
```
But I am wondering, is this restriction on the notebook version really necessary?

And any ideas to overcome it? I have seen [this Github issue](https://github.com/OpenMined/PySyft/issues/5315
), but it seems still open

**Comment by desertnaut** [comment]
Although the relevant [Github issue](https://github.com/OpenMined/PySyft/issues/5315) now appears closed, it leads to a different error when installing pysyft according to the course instructions:
```python
!pip install git+https://github.com/OpenMined/PySyft@dev#egg=syft
!git clone https://github.com/OpenMined/courses.git --branch foundations-of-private-computation --single-branch
import syft as sy

---------------------------------------------------------------------------

AttributeError                            Traceback (most recent call last)

/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py in _dep_map(self)
   3015         try:
-> 3016             return self.__dep_map
   3017         except AttributeError:

14 frames

AttributeError: _DistInfoDistribution__dep_map


During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)

AttributeError: _pkg_info


During handling of the above exception, another exception occurred:

FileNotFoundError                         Traceback (most recent call last)

/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py in _get(self, path)
   1609 
   1610     def _get(self, path):
-> 1611         with open(path, 'rb') as stream:
   1612             return stream.read()
   1613 

FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/lib/python3.7/dist-packages/torchvision-0.9.0+cu101.dist-info/METADATA'
```

**Comment by znreza** [comment]
Hi @desertnaut there is a [special script](https://github.com/OpenMined/PySyft/blob/dev/scripts/colab.sh) and prelude that's needed to install syft in Colab. You can give it a try. However, let me tell you that Colab is not still fully functional with syft at the moment. So even if you could install everything, you may still face some connection issue between the duet servers.

**Reply by ss007k** [reply]
I hit the same issue; asked on Slack before coming here to check.
Do you know if this could get fixed in, say, 2 weeks from now? Lot of folks use Colab, so it would really help to get it working there. (personally my laptop is bit old, and hence an online service helps).

I could manage to install syft but could not import it:

---------------------------------------------------------------------------
ContextualVersionConflict                 Traceback (most recent call last)
<ipython-input-7-281ee5c8ca5f> in <module>()
----> 1 import syft as sy
4 frames
/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py
 in resolve(self, requirements, env, installer, replace_conflicting, extras)
    775                 # Oops, the "best" so far conflicts with a dependency
    776                 dependent_req = required_by[req]
--> 777                 raise VersionConflict(dist, req).with_context(dependent_req)
    778 
    779             # push the new requirements onto the stack
ContextualVersionConflict: (torchvision 0.9.1+cu101 (/usr/local/lib/python3.7/dist-packages), Requirement.parse('torchvision<=0.9,>=0.5'), {'syft'})

**Comment by omar-ads** [comment]
!pip install torchvision==0.8.2
!pip install syft==0.3.0

this solved the issue for me and the code run on Colab just fine

hope it helps

**Reply by desertnaut** [reply]
this was not what the question was about; the course requires installing the specific dev version of pysyft

**Reply by omar-ads** [reply]
Ohh sorry i didn’t read the question fully and thought you had the same problem because i was following the split learning notebooks and got an error when it ran import syft as sy
Don’t mind my answer then 🙏🏼

**Reply by desertnaut** [reply]
No worries; thanks for trying.

**Comment by Suni-Jose** [comment]
I got the error  no module named syft when i tried to use it in google colab.How to fix it

**Reply by Aarti0199** [reply]
You can install syft by using !pip install syft.
I was able to solve issue using this, Hope this will solve your problem as well.

**Comment by ashiqnazirbhat** [comment]
NameError: name 'RequestMessage' is not defined

Getting this error for syft==0.3.0
